src/main/java/programminglife/parser/GraphParser.java

```diff
@@ -108,14 +105,14 @@ protected synchronized void parse(boolean verbose) throws IOException, UnknownTy
                         this.parseLink(line);
                         break;
                     case 'H':
-                        System.out.println(line);
+                        this.parseHeader(line);
                         break;
                     default:
                         throw new UnknownTypeException(String.format("Unknown symbol '%c'", type));
                 }
 
                 if (Thread.currentThread().isInterrupted()) {
```

Comment:
I don't think this does what you want, you should probably catch a InterruptedException, rollback, and then rethrow.